File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/EventItem.java

package com.noblemajesty.marvel.models.getCharacters;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class EventItem implements Parcelable
{

  @SerializedName("resourceURI")
  @Expose
  private String resourceURI;
  @SerializedName("name")
  @Expose
  private String name;
  public final static Parcelable.Creator<EventItem> CREATOR = new Creator<EventItem>() {


    @SuppressWarnings({
        "unchecked"
    })
    public EventItem createFromParcel(Parcel in) {
      return new EventItem(in);
    }

    public EventItem[] newArray(int size) {
      return (new EventItem[size]);
    }

  }
      ;

  protected EventItem(Parcel in) {
    this.resourceURI = ((String) in.readValue((String.class.getClassLoader())));
    this.name = ((String) in.readValue((String.class.getClassLoader())));
  }

  public EventItem() {
  }

  public String getResourceURI() {
    return resourceURI;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(resourceURI);
    dest.writeValue(name);
  }

  public int describeContents() {
    return 0;
  }

}

File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/EventItem.kt
package com.noblemajesty.marvel.models.getCharacters

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class EventItem : Parcelable {

    @SerializedName("resourceURI")
    @Expose
    var resourceURI: String = ""
    @SerializedName("name")
    @Expose
    var name: String? = null

    protected constructor(`in`: Parcel) {
        this.resourceURI = `in`.readValue(String::class.java.classLoader) as String
        this.name = `in`.readValue(String::class.java.classLoader) as String
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(resourceURI)
        dest.writeValue(name)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<EventItem> = object : Parcelable.Creator<EventItem> {


            override fun createFromParcel(`in`: Parcel): EventItem {
                return EventItem(`in`)
            }

            override fun newArray(size: Int): Array<EventItem?> {
                return arrayOfNulls(size)
            }

        }
    }

}
